include/psen_scan_v2/scanner_v2.h

```diff
@@ -72,26 +74,31 @@ class ScannerV2 : public IScanner
   void scannerStartedCB();
   void scannerStoppedCB();
 
-  void startStartWatchdog();
-  void stopStartWatchdog();
+private:
+  using OptionalPromise = boost::optional<std::promise<void>>;
 
 private:
-  std::promise<void> scanner_has_started_;
-  std::promise<void> scanner_has_stopped_;
+  OptionalPromise scanner_has_started_{ boost::none };
+  OptionalPromise scanner_has_stopped_;
+
+  //! @brief This Mutex protects ALL members of the Scanner against concurrent access.
+  //! So far there exist at least the following threads, potentially causing concurrent access to the members:
+  //! - user-main-thread
+  //! - io_service thread of UDPClient
+  //! - watchdog threads
+  std::recursive_mutex member_mutex_;
```

Comment:
I don't see the need for a recursive mutex. So I would use `std::mutex`.

Reply:
No, its needed because of the `triggerEventWithParam()` function. The only other option would be to use an internal version of the `triggerEventWithParam()` function which not automatically locks the mutex.

Reply:
I tried it with `std::mutex` and using `sm_->process_event(event);` directly  (in other words without taking the mutex). It does not work, probably because a call to  `sm_->process_event(event);` might cause callbacks to be called which, in turn, also want lock the mutex. In this case the program, consequently, ends in a deadlock.

Reply:
Guess by tracking the locks the need for a recursive mutex should be easy to verify